src/imageio/format/jxl.c

```diff
@@ -564,76 +548,61 @@ void gui_init(dt_imageio_module_format_t *self)
   self->widget = box;
 
   // bits per sample combobox
-  GtkWidget *bpp = dt_bauhaus_combobox_new(NULL);
-  dt_bauhaus_combobox_add(bpp, _("8 bit"));
-  dt_bauhaus_combobox_add(bpp, _("10 bit"));
-  dt_bauhaus_combobox_add(bpp, _("12 bit"));
-  dt_bauhaus_combobox_add(bpp, _("16 bit"));
-  dt_bauhaus_combobox_add(bpp, _("16 bit (half)"));
-  dt_bauhaus_combobox_add(bpp, _("32 bit (float)"));
   const int bpp_enum = dt_conf_get_int("plugins/imageio/format/jxl/bpp");
-  dt_bauhaus_combobox_set(bpp, bpp_enum);
-  dt_bauhaus_widget_set_label(bpp, NULL, N_("bit depth"));
-  g_signal_connect(G_OBJECT(bpp), "value-changed", G_CALLBACK(bpp_changed), self);
-  gtk_box_pack_start(GTK_BOX(box), bpp, TRUE, TRUE, 0);
-  gui->bpp = bpp;
+
+  DT_BAUHAUS_COMBOBOX_NEW_FULL(gui->bpp, self, NULL, N_("bit depth"), NULL, bpp_enum, bpp_changed, gui,
+                               N_("8 bit"), N_("10 bit"), N_("12 bit"), N_("16 bit"), N_("16 bit (half)"),
+                               N_("32 bit (float)"));
+  gtk_box_pack_start(GTK_BOX(box), gui->bpp, TRUE, TRUE, 0);
 
   // quality slider
-  GtkWidget *quality
-      = dt_bauhaus_slider_new_with_range(NULL, dt_confgen_get_int("plugins/imageio/format/jxl/quality", DT_MIN),
-                                         dt_confgen_get_int("plugins/imageio/format/jxl/quality", DT_MAX), 1,
-                                         dt_confgen_get_int("plugins/imageio/format/jxl/quality", DT_DEFAULT), 0);
-  const int quality_val = dt_conf_get_int("plugins/imageio/format/jxl/quality");
-  dt_bauhaus_slider_set(quality, quality_val);
-  dt_bauhaus_widget_set_label(quality, NULL, _("quality"));
-  gtk_widget_set_tooltip_text(quality, _("the quality of the output image\n0-29 = very lossy\n30-99 = JPEG "
-                                         "quality comparable\n100 = lossless (integer bith depth only)"));
-  g_signal_connect(G_OBJECT(quality), "value-changed", G_CALLBACK(quality_changed), self);
-  gtk_box_pack_start(GTK_BOX(box), quality, TRUE, TRUE, 0);
-  gui->quality = quality;
+  const int quality = dt_conf_get_int("plugins/imageio/format/jxl/quality");
+
+  gui->quality = dt_bauhaus_slider_new_with_range(
+      (dt_iop_module_t *)self, dt_confgen_get_int("plugins/imageio/format/jxl/quality", DT_MIN),
+      dt_confgen_get_int("plugins/imageio/format/jxl/quality", DT_MAX), 1,
+      dt_confgen_get_int("plugins/imageio/format/jxl/quality", DT_DEFAULT), 0);
+  dt_bauhaus_slider_set(gui->quality, quality);
+  dt_bauhaus_widget_set_label(gui->quality, NULL, _("quality"));
```

Comment:
`dt_bauhaus_widget_set_label` should be fed untranslated names (like `N_("quality")`) so that shortcut definitions don't break when the language is changed. Same below.

Reply:
Thanks for looking over, will correct in the next push.